File: ascii_to_hex.js
function asciiToHex(ascii) {
  var hex = "";
  for (var i = 0; i < ascii.length; i++) {
    hex += String(ascii.charCodeAt(i)).toString(16).padStart(2, '0');
  }
  return hex;
}

// Example
var ascii = "Hello, World!";
var hex = asciiToHex(ascii);
console.log("The hexadecimal representation of " + ascii + " is " + hex);

File: ascii_to_hex.swift
func asciiToHexadecimal(ascii: String) -> String {
  var hexadecimal = ""
  for char in ascii {
    hexadecimal += String(format: "%02x", char)
  }
  return hexadecimal
}

let ascii = "Hello, World!"
let hexadecimal = asciiToHexadecimal(ascii)
print("The hexadecimal representation of \(ascii) is \(hexadecimal)")

